Resources/doc/middlewares.rst

```diff
@@ -0,0 +1,90 @@
+Middlewares
+===========
+
+Doctrine DBAL supports middlewares. According to the `Doctrine documentation`_:
+
+    "A middleware sits in the middle between the wrapper components and the driver"
+
+They allows to decorates the following DBAL classes:
+
+- Doctrine\\DBAL\\Driver
+- Doctrine\\DBAL\\Driver\\Connection
+- Doctrine\\DBAL\\Driver\\Statement
+- Doctrine\\DBAL\\Driver\\Result
+
+Symfony, for instance, uses a middleware to harvest the queries executed by the current page and make them available in the profiler.
+
+.. _`Doctrine documentation`: https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/architecture.html#middlewares
+
+You can also create your own middleware. This is an example of a (very) simple middleware that prevent database connection with the root user.
+The first step is to create the middleware:
+
+.. code-block:: php
+
+    <?php
+
+    namespace App\Middleware;
+
+    use Doctrine\DBAL\Driver;
+    use Doctrine\DBAL\Driver\Middleware;
+
+    class PreventRootConnectionMiddleware implements Middleware
+    {
+        public function wrap(Driver $driver): Driver
+        {
+            return new PreventRootConnectionDriver($driver);
+        }
+    }
+
+As you can see in the ``wrap`` method, middlewares principle is to decorate Doctrine objects with your own objects bearing the logic you need. Now, the connect method of the driver must be decorated in ``PreventRootConnectionDriver`` to prevent connection with the root user:
```

Comment:
```suggestion
As you can see in the ``wrap`` method, the principle of a middleware is to decorate Doctrine objects with your own objects bearing the logic you need. Now, the ``connect`` method of the driver must be decorated in ``PreventRootConnectionDriver`` to prevent connections with the root user:
```